src/EventListener/InitializeSystemListener.php

```diff
@@ -319,29 +324,43 @@ private function validateInstallation(Request $request = null)
         }
 
         // Show the "insecure document root" message
-        // FIXME: add unit tests for this as soon as die_nicely is an exception
         if (!in_array($request->getClientIp(), ['127.0.0.1', 'fe80::1', '::1']) && '/web' === substr($request->getBasePath(), -4)) {
-            die_nicely('be_insecure', 'Your installation is not secure. Please set the document root to the <code>/web</code> subfolder.');
+            throw new InsecureInstallationHttpException(
+                null,
+                'Your installation is not secure. Please set the document root to the <code>/web</code> subfolder.'
+            );
         }
 
         // Show the "incomplete installation" message
         if (!$this->config->isComplete()) {
-            die_nicely('be_incomplete', 'The installation has not been completed. Open the Contao install tool to continue.');
+            // FIXME: This maybe should be removed when localconfig.php is not mandatory anymore.
+            throw new IncompleteInstallationHttpException(
+                null,
+                'The installation has not been completed. Please finish the configuration.'
+            );
         }
     }
 
     /**
+     * Set the configured error level.
+     */
+    private function sanitizeErrorHandling()
+    {
+        error_reporting($this->container->getParameter('contao.error_level'));
+    }
+
+    /**
      * Configures the error handling.
      */
     private function configureErrorHandling()
     {
         // Always show error messages if logged into the install tool (see #5001)
+        // FIXME: remove when the install tool is gone.
         if (Input::cookie('TL_INSTALL_AUTH') && !empty($_SESSION['TL_INSTALL_AUTH']) && Input::cookie('TL_INSTALL_AUTH') == $_SESSION['TL_INSTALL_AUTH'] && $_SESSION['TL_INSTALL_EXPIRE'] > time()) {
             $this->config->set('displayErrors', 1);
         }
 
         $this->iniSet('display_errors', ($this->config->get('displayErrors') ? 1 : 0));
-        error_reporting(($this->config->get('displayErrors') || $this->config->get('logErrors')) ? $this->config->get('errorReporting') : 0);
```

Comment:
It is set twice if any of the flags is specified. It get's reset to 0 otherwise which is of even more problematic nature as it is effectively killing the complete error handling of symfony, in all environments, I doubt that this is intended (it even corrupted the unit tests on my side which is actually the reason I dropped the setting of 0).
IMO we could drop the whole method as the logging should get redirected to the app/logs no matter what the displayErrors or logErrors flags of Contao say (logging is still an upcoming PR) and therefore honor what the application config says instead of arbitrarily modding the handlers.
I'd accept it if this would be restricted to the install tool - which is scheduled for removal anyway - but this modifies the behaviour of the whole application in a manner that can not be configured in the symfony way - the `contao.error_level` is already a hack I dislike but it is at least still in symfony environment.